### PR TITLE
Issue 1700 - Prevent hzn user from accidentally creating deployment p…

### PR DESCRIPTION
**Pull request by linggao**
…olicy with no constraints

### CODE REVIEW

File: cli/hzn.go [hunk 1]
```diff
@@ -296,6 +296,7 @@ Environment Variables:
 	exBusinessAddPolicyIdTok := exBusinessAddPolicyCmd.Flag("id-token", msgPrinter.Sprintf("The Horizon ID and password of the user.")).Short('n').PlaceHolder("ID:TOK").String()
 	exBusinessAddPolicyPolicy := exBusinessAddPolicyCmd.Arg("policy", msgPrinter.Sprintf("The name of the policy to add or overwrite.")).Required().String()
 	exBusinessAddPolicyJsonFile := exBusinessAddPolicyCmd.Flag("json-file", msgPrinter.Sprintf("The path of a JSON file containing the metadata necessary to create/update the service policy in the Horizon Exchange. Specify -f- to read from stdin.")).Short('f').Required().String()
+	exBusinessAddPolNoConstraint := exBusinessAddPolicyCmd.Flag("no-constraints", msgPrinter.Sprintf("This deployment policy does not have constraints.")).Bool()
```
Comment (dabooz) [inline cli/hzn.go:299]: Please reword the help to "Allow this deployment policy to be published even though it does not have any constraints."

File: cli/exchange/business.go [hunk 2]
```diff
@@ -86,6 +86,17 @@ func BusinessAddPolicy(org string, credToUse string, policy string, jsonFilePath
 		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("Incorrect deployment policy format in file %s: %v", jsonFilePath, err))
 	}
 
+	cons := len(policyFile.Constraints)
+	if noConstraints {
+		if cons > 0 {
+			cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("--no-constraints flag is specified but the given deployment policy contains %v constraints.", cons))
```
Comment (dabooz) [inline cli/exchange/business.go:92]: I dont think we should give an error in this case.

File: cli/exchange/business.go [hunk 3]
```diff
@@ -86,6 +86,17 @@ func BusinessAddPolicy(org string, credToUse string, policy string, jsonFilePath
 		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("Incorrect deployment policy format in file %s: %v", jsonFilePath, err))
 	}
 
+	cons := len(policyFile.Constraints)
```
Comment (dabooz) [inline cli/exchange/business.go:89]: Is checking the length of the constraints array enough? What if all the elements in the array are the empty string?

File: cli/exchange/business.go [hunk 4]
```diff
@@ -86,6 +86,17 @@ func BusinessAddPolicy(org string, credToUse string, policy string, jsonFilePath
...
+	if noConstraints {
+		if cons > 0 {
+			cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("--no-constraints flag is specified but the given deployment policy contains %v constraints.", cons))
+		}
+	} else {
+		if cons == 0 {
+			cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("You must use --no-constraints flag because the the given deployment policy does not contain any constraints."))
```
Comment (dabooz) [inline cli/exchange/business.go:96]: Please change the error message to: "The deployment policy has no constraints which might result in the service being deployed to all nodes. Please specify --no-constraints to confirm that this is acceptable."

File: cli/hzn.go (hunk 1 above)
Comment (linggao) [reply cli/hzn.go:299]: done

File: cli/exchange/business.go (hunk 2 above)
Comment (linggao) [reply cli/exchange/business.go]: done

File: cli/exchange/business.go (hunk 3 above)
Comment (linggao) [reply cli/exchange/business.go]: Sounds over protecting. But I did it.

File: cli/exchange/business.go (hunk 4 above)
Comment (linggao) [reply cli/exchange/business.go]: Done

**Pull request merged**